stability: panic on delta in ChangeReplicas

three nodes on delta (13d3b39-dirty) crashed with:

```
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xb17d66]

goroutine 232 [running]:
panic(0x1751040, 0xc42000e070)
        /usr/local/go/src/runtime/panic.go:500 +0x1a1
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).Recover(0xc42043e5a0)
        /go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:185 +0x6e
panic(0x1751040, 0xc42000e070)
        /usr/local/go/src/runtime/panic.go:458 +0x243
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).Recover(0xc42043e5a0)
        /go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:185 +0x6e
panic(0x1751040, 0xc42000e070)
        /usr/local/go/src/runtime/panic.go:458 +0x243
github.com/cockroachdb/cockroach/pkg/storage.(*Replica).ChangeReplicas.func1.1(0x7f35a2100980, 0xc42f7ffce0, 0x0, 0xc424948800)
        /go/src/github.com/cockroachdb/cockroach/pkg/storage/replica_command.go:3181 +0x26
github.com/cockroachdb/cockroach/pkg/storage.(*Replica).ChangeReplicas.func1(0xc424948800, 0x7f35a2100980, 0xc42f7ffce0, 0xc42403ce04, 0xc42cc50120, 0x23e55e0, 0xc4294da2a0)
        /go/src/github.com/cockroachdb/cockroach/pkg/storage/replica_command.go:3186 +0x1c8
github.com/cockroachdb/cockroach/pkg/storage.(*Replica).ChangeReplicas(0xc424948800, 0x7f35a2100980, 0xc42f7ffce0, 0x400000000, 0x8, 0xc430638300, 0x0, 0x0)
        /go/src/github.com/cockroachdb/cockroach/pkg/storage/replica_command.go:3230 +0x3d9
github.com/cockroachdb/cockroach/pkg/storage.(*replicateQueue).addReplica(0xc420440540, 0x7f35a2100980, 0xc42f7ffce0, 0xc424948800, 0x800000004, 0xc400000000, 0xc430638300, 0x1, 0x0)
        /go/src/github.com/cockroachdb/cockroach/pkg/storage/replicate_queue.go:322 +0x79
github.com/cockroachdb/cockroach/pkg/storage.(*replicateQueue).process(0xc420440540, 0x7f35a2100980, 0xc42f7ffce0, 0x1488f1ccc612fb0a, 0x0, 0xc424948800, 0xc426344800, 0x17, 0x20, 0xc42e715bd0, ...)
        /go/src/github.com/cockroachdb/cockroach/pkg/storage/replicate_queue.go:306 +0xc1b
github.com/cockroachdb/cockroach/pkg/storage.(*baseQueue).processReplica(0xc4209b41e0, 0x7f35a2145c80, 0xc42e5f2270, 0xc424948800, 0xc42052cb40, 0x0, 0x0)
        /go/src/github.com/cockroachdb/cockroach/pkg/storage/queue.go:548 +0x43c
github.com/cockroachdb/cockroach/pkg/storage.(*baseQueue).processLoop.func1.2()
        /go/src/github.com/cockroachdb/cockroach/pkg/storage/queue.go:473 +0xa2
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunTask(0xc42043e5a0, 0xc42c04fe68, 0x0, 0x0)
        /go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:224 +0x100
github.com/cockroachdb/cockroach/pkg/storage.(*baseQueue).processLoop.func1()
        /go/src/github.com/cockroachdb/cockroach/pkg/storage/queue.go:477 +0x321
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc42043e5a0, 0xc4203462c0)
        /go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:196 +0x7d
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
        /go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:197 +0x66
```

#10875 tweaks some of this code, but shows L3181 as `log.Event(ctx, "generated snapshot")` which seems unlikely to be the culprit.

--- a/pkg/storage/replica_command.go
+++ b/pkg/storage/replica_command.go
@@ -3177,11 +3177,11 @@ func (r *Replica) ChangeReplicas(
 		// complete. See #10409.
 		if err := func() error {
 			snap, err := r.GetSnapshot(ctx, snapTypePreemptive)
+			if err != nil {
+				return errors.Wrapf(err, "%s: change replicas failed to generate snapshot", r)
+			}
 			defer r.CloseOutSnap()
 			log.Event(ctx, "generated snapshot")
-			if err != nil {
-				return errors.Wrapf(err, "%s: change replicas failed", r)
-			}
 
 			fromRepDesc, err := r.GetReplicaDescriptor()
 			if err != nil {

